utilities/unittests/test_guest_support.py

```
@@ -510,3 +514,75 @@ def side_effect_generator():
+
+        mock_start_and_fetch_pid.assert_called_once_with(vm=mock_vm, process_name=OS_PROC_NAME["windows"])
+        mock_pause_unpause.assert_called_once_with(vm=mock_vm)
+        mock_fetch_pid.assert_called_once_with(vm=mock_vm, process_name=OS_PROC_NAME["windows"])
+        mock_kill_processes.assert_called_once_with(vm=mock_vm, process_name=OS_PROC_NAME["windows"])
+
+    @patch("utilities.guest_support.kill_processes_by_name_windows")
+    @patch("utilities.guest_support.fetch_pid_from_windows_vm")
+    @patch("utilities.guest_support.pause_unpause_vm_and_check_connectivity")
+    @patch("utilities.guest_support.start_and_fetch_processid_on_windows_vm")
+    def test_validate_pause_unpause_windows_vm_uses_provided_pre_pause_pid(
+        self, mock_start_and_fetch_pid, mock_pause_unpause, mock_fetch_pid, mock_kill_processes
```

Ruff (0.15.21)

[warning] Unused method argument: `mock_kill_processes`

(ARG002)